mls_validation_service/src/handlers.rs

```diff
@@ -81,6 +86,51 @@ impl ValidationApi for ValidationService {
             responses: out,
         }))
     }
+
+    async fn get_association_state(
+        &self,
+        request: Request<GetAssociationStateRequest>,
+    ) -> Result<Response<GetAssociationStateResponse>, Status> {
+        let GetAssociationStateRequest {
+            old_updates,
+            new_updates,
+        } = request.into_inner();
+
+        let conv_proto =
+            |updates: Vec<IdentityUpdateProto>| -> Result<Vec<IdentityUpdate>, Status> {
+                updates
+                    .into_iter()
+                    .map(IdentityUpdate::from_proto)
+                    .collect::<Result<Vec<_>, _>>()
+                    .map_err(|e| Status::new(Code::InvalidArgument, e.to_string()))
+            };
+
+        let to_status = |e: AssociationError| Status::new(Code::InvalidArgument, e.to_string());
```

Comment:
Seems like this could live outside the function, since it doesn't require any context

Reply:
removed both closures